packages/react-dom/src/__tests__/ReactDOMNativeEventHeuristic-test.js

```diff
@@ -281,4 +281,44 @@ describe('ReactDOMNativeEventHeuristic-test', () => {
       expect(container.textContent).toEqual('hovered');
     });
   });
+
+  // @gate experimental
+  it('should batch inside native events', async () => {
+    const root = ReactDOM.unstable_createRoot(container);
+
+    const target = React.createRef(null);
+    function Foo() {
+      const [count, setCount] = React.useState(0);
+
+      React.useEffect(() => {
+        Scheduler.unstable_yieldValue(count);
+      }, [count]);
+
+      React.useLayoutEffect(() => {
+        target.current.onclick = () => {
+          setCount(c => c + 1);
```

Comment:
If you change these to `setCount(c + 1)` instead of using the updater then the final DOM output is different depending on whether they are batched. 1 if batched, 2 if not batched.

Reply:
Just a suggestion because the test failure will be more obvious. Like if you ran this in a sandbox, you wouldn't notice if there were two synchronous commits instead of one, but you would notice that it's displaying the wrong number.